src/proxy.rs

```diff
@@ -873,41 +890,115 @@ fn is_cgi() -> bool {
+        .and_then(|flag| flag.to_i32())
+        .unwrap_or(0)
+        == 1;
+
+    if proxy_enabled {
+        let proxy_host = proxies_map
+            .find(host_key)
+            .and_then(|host| host.downcast::<CFString>())
+            .map(|host| host.to_string())
+            .unwrap_or_default();
+        let proxy_port = proxies_map
+            .find(port_key)
+            .and_then(|port| port.downcast::<CFNumber>())
+            .and_then(|port| port.to_i32());
+
+        if let Some(proxy_port) = proxy_port {
+            return Some(format_args!("{scheme}={proxy_host}:{proxy_port}").to_string());
+        } else {
+            return Some(format_args!("{scheme}={proxy_host}").to_string());
+        }
+    }
 
-#[cfg(not(target_os = "windows"))]
-fn get_from_registry() -> Option<RegistryProxyValues> {
     None
 }
 
-#[cfg(target_os = "windows")]
-fn parse_registry_values_impl(
-    registry_values: RegistryProxyValues,
-) -> Result<SystemProxyMap, Box<dyn Error>> {
-    let (proxy_enable, proxy_server) = registry_values;
+#[cfg(target_os = "macos")]
+fn get_from_platform_impl() -> Result<Option<String>, Box<dyn Error>> {
+    let store = SCDynamicStoreBuilder::new("reqwest").build();
+
+    if let Some(proxies_map) = store.get_proxies() {
+        let http_proxy_config = parse_setting_from_dynamic_store(
+            &proxies_map,
+            unsafe { kSCPropNetProxiesHTTPEnable },
+            unsafe { kSCPropNetProxiesHTTPProxy },
+            unsafe { kSCPropNetProxiesHTTPPort },
+            "http",
+        );
+        let https_proxy_config = parse_setting_from_dynamic_store(
+            &proxies_map,
+            unsafe { kSCPropNetProxiesHTTPSEnable },
+            unsafe { kSCPropNetProxiesHTTPSProxy },
+            unsafe { kSCPropNetProxiesHTTPSPort },
+            "https",
+        );
 
-    if proxy_enable == 0 {
-        return Ok(HashMap::new());
+        match (http_proxy_config, https_proxy_config) {
+            (Some(http_proxy_config), Some(https_proxy_config)) => {
+                return Ok(Some(
+                    format_args!("{http_proxy_config};{https_proxy_config}").to_string(),
+                ))
+            }
+            (Some(config), None) | (None, Some(config)) => {
+                return Ok(Some(config));
+            }
+            (None, None) => {
+                return Ok(None);
+            }
+        }
+    }
+
+    Ok(None)
+}
+
+#[cfg(any(target_os = "windows", target_os = "macos"))]
+fn get_from_platform() -> Option<String> {
+    if let Some(platform_proxy_values) = get_from_platform_impl().ok() {
+        platform_proxy_values
+    } else {
+        None
     }
+}
+
+#[cfg(not(any(target_os = "windows", target_os = "macos")))]
+fn get_from_platform() -> Option<String> {
+    None
+}
 
+#[cfg(any(target_os = "windows", target_os = "macos"))]
+fn parse_platform_values_impl(platform_values: String) -> Result<SystemProxyMap, Box<dyn Error>> {
```

Comment:
Could you clarify why this returns a `Result<SystemProxyMap>`? I don't believe we ever return an `Err`

Reply:
I'm not entirely sure myself. I just changed this to return a `SystemProxyMap`, and got rid of the unneeded `.unwrap_or(HashMap::new())` in the non-impl version.